docs/Exit-Codes.md

```diff
@@ -0,0 +1,266 @@
+
+| Exit&nbsp;Code | Reason |
+|:-----|----------|
+| 0 | Success |
+| [70](#70) | Unexpected internal software issue. |
+| [73](#73) | Can't create output file. |
+| [100](#100) | Instantiation Failed - Processing issues. |
+| [101](#101) | Template authoring issues. |
+| [102](#102) | Missing required option(s) and/or argument(s) for the command. |
+| [103](#103) | The template or the template package was not found. |
+| [104](#104) | PostAction operation was cancelled. |
+| [105](#105) | Instantiation Failed - Post action failed. |
+| [106](#106) | Template/Package management operation Failed. |
+| [107 - 113](#107) | Reserved. |
+| [127](#127) | Unrecognized option(s) and/or argument(s) for a command. |
+| [130](#130) | Command terminated by user. |
+
+
+To enable verbose logging in order to troubleshoot issue(s), set the `DOTNET_CLI_CONTEXT_VERBOSE` environment variable to `true`
+
+_PowerShell:_
+```PowerShell
+$env:DOTNET_CLI_CONTEXT_VERBOSE = 'true'
+```
+
+_Cmd:_
+```cmd
+set DOTNET_CLI_CONTEXT_VERBOSE=true
+```
+
+## <a name="70"></a>70 - Unexpected internal software issue
+
+Unexpected result or issue. [File a bug](https://github.com/dotnet/templating/issues/new?title=Unexpected%20Internal%20Software%20Issue%20(EX_SOFTWARE)) if you encounter this exit code.
+
+This is a semi-standardized exit code (see [EX_SOFTWARE in /usr/include/sysexits.h](https://github.com/openbsd/src/blob/master/include/sysexits.h#L107))
+
+
+## <a name="73"></a>73 - Can't create output file.
+
+The operation was cancelled due to detection of an attempt to perform destructive changes to existing files. This can happen if you are attempting to instantiate template into the same folder where it was previously instantiated under same target name (specified via `--name` option or defaults to the target directory name)
+
+_Example:_
+```console
+> dotnet new console
+
+The template "Console App" was created successfully.
+
+Processing post-creation actions...
+Running 'dotnet restore' on C:\tmp\tmp.csproj...
+  Determining projects to restore...
+  Restored C:\tmp\tmp.csproj (in 47 ms).
+Restore succeeded.
+
+> dotnet new console
+
+Creating this template will make changes to existing files:
+  Overwrite   ./tmp.csproj
+  Overwrite   ./Program.cs
+
+Rerun the command and pass --force to accept and create.
+
+For details on current exit code please visit https://aka.ms/templating-exit-codes#73
+```
+
+Destructive changes can be forced by passing `--force` option
+
+This is a semi-standardized exit code (see [EX_CANTCREAT in /usr/include/sysexits.h](https://github.com/openbsd/src/blob/master/include/sysexits.h#L110))
+
+
+## <a name="100"></a>100 - Instantiation Failed - Processing issues
+
+The template instantiation failed due to error(s). Caused by environment (failure to read/write template(s) or cache) or 
+
+## <a name="101"></a>101 - Template authoring issues
```

Comment:
do we plan to use it also during validation on install? or will it be only for instantiation?
`Invalid template or template package` might be a better category.